Revise the solution of rekeying the initial Child SA (#11)
William-PanWei/draft-ietf-ipsecme-ikev2-sa-ts-payloads-opt: draft-ietf-ipsecme-ikev2-sa-ts-payloads-opt.md

                                         Nr, [KEr,]}
~~~~

For the initial Child SA that was negotiated as part of an initial IKE exchange (e.g., IKE_AUTH), at the time of its creation the parameters of PFS and KE method for Child SAs are not negotiated.

> Therefore, when rekeying the initial Child SA for the first time, a regular rekey MUST be used for the negotiation of these parameters. Then, the next rekey can use the optimized way.

Reviewer: i don't want to unconditionally say that, as i am still hoping to kill the regular rekey.

Author: I've revised the text again. Now it highlights that the optimized rekey can be used if draft-child-pfs-info is used. You can have a check of the new text.

                                         Nr, [KEr,]}
~~~~

> If a responder fails to process the optimized rekey request, e.g., there is a change of the configuration, it MUST return an error as the notification of type NO_PROPOSAL_CHOSEN. After receiving the error response of the optimized rekey, the initiator can retry a regular rekey.

Reviewer 2: This is a bit vague. What if responder fails to process for some other reason, e.g. it cannot create more SAs? I think we should be more specific: "If a responder fails to process the optimized rekey request because for some reason it cannot re-use SA parameters for the SA being re-reyed (e.g., there is a change in the responder's configuration), ..." or something like this.

Author: ```suggestion
If a responder fails to process the optimized rekey request because for some reasons it cannot re-use SA parameters for the SA being rekeyed (e.g., there is a change in the responder's configuration), it MUST return an error as the notification of type NO_PROPOSAL_CHOSEN. After receiving the error response of the optimized rekey, the initiator can retry a regular rekey.
```

## Optimized Rekey of the Initial Child SA

For the initial Child SA that was negotiated as part of an initial IKE exchange (e.g., IKE_AUTH), at the time of its creation the parameters of PFS and KE method(s) for Child SAs are not negotiated.
However, {{-child-pfs}} provides a mechanism to negotiate these parameters during the creation of the initial Child SA.

If both peers support and use {{-child-pfs}}, the PFS policy and KE method(s) for the initial Child SA is known during its creation.

> Therefore, when rekeying the initial Child SA for the first time, the optimized way SHOULD be used.

Reviewer 2: I woud add: "Therefore, in this situation, ..." to emhasizi that both peer should know KE method(s).

Author: ```suggestion
Therefore, in this situation, when rekeying the initial Child SA for the first time, the optimized way SHOULD be used.
```

Some implementations allow sending rekey messages with a different set of Traffic Selectors or cryptographic parameters in response to a configuration update. IKEv2 [RFC7296] states this "SHOULD NOT" be done. But if there is a configuration change that changes the Traffic Selectors, cryptographic parameters, or other properties of the SA, the regular rekey should be used to make the configuration change active, since the optimized rekey can't express such changes.

> Two peers' PFS policy and KE method configurations MUST be the same, otherwise, the rekey (regardless of regular or optimized way) of the initial Child SA created in the IKE_AUTH exchange would fail. This issue is also discussed in detail in {{-child-pfs}}. 

Reviewer 2: Why configurations "MUST be the same"? They MUST contain mutually accepatable KE methods, but they need not be the same.

Author: ```suggestion
Two peers MUST have the same PFS policy and contain mutally acceptable KE method(s), otherwise, the rekey (regardless of regular or optimized way) of the initial Child SA created in the IKE_AUTH exchange would fail. This issue is also discussed in detail in {{-child-pfs}}. 
```

Reviewer: I think "the same" refers to "the same list or options", so I think it is okay.
If they were truly not the same list, you would get different behaviour based on being initiator or responder, which would be a fairly bogus policy to allow.
